.github/workflows/build_and_deploy.yml

```
@@ -289,7 +288,7 @@ jobs:
         env:
           ORT_USE_CUDA: ${{ matrix.use_cuda }}
       - name: build voicevox_core_java_api
-        if: contains(matrix.target, 'android')
+        if: "!contains(matrix.target, 'ios')"
```

文字列にしないで良い気がします。でも最初に ! があるとYAML的に別の意味を持っちゃう可能性があるんですね。。
調べてみた感じ

> If the condition statement is started with “!” (Not), the expression syntax “${{ }}” can’t be omitted.

とのことなので、こうですかね？
https://github.com/orgs/community/discussions/26386#discussioncomment-3251726

```suggestion
        if: ${{ !contains(matrix.target, 'ios') }}
```